### PR TITLE
refactor(utils): Remove unused utils

### PR DESCRIPTION
**Pull request by provinzkraut**
Remove unused utils

### CODE REVIEW

**Review by JacobCoffee — APPROVED** [review]
what if someone is using these out in the wild? These don't seem privatized

**Comment by provinzkraut** [comment]
> what if someone is using these out in the wild? These don't seem privatized

They are not, they're just "implicitly private". We should move them to a private module. At the moment they are simply intentionally undocumented because we have been treating them as if they were private; We usually don't care about backwards compatibility there, don't advertise their usage in the docs and they are missing from the reference docs.

I've put up a discussion on discord about this already but feel free to share your thoughts here as well.

File: tests/unit/test_utils/test_typing.py [hunk 1]
```diff
@@ -68,93 +40,3 @@ def test_get_origin_or_inner_type() -> None:
...
-            {
-                "bound_foo": BoundFoo[int],
-                "constrained_foo": ConstrainedFoo[Union[int, str]],  # type: ignore[type-var]
-                "constrained_foo_with_t": ConstrainedFoo[int],
-            },
-        ),
-    ),
-)
-def test_get_type_hints_with_generics(annotation: Any, expected_type_hints: dict[str, Any]) -> None:
-    assert get_type_hints_with_generics_resolved(annotation, include_extras=True) == expected_type_hints
```
Comment (guacs) [inline tests/unit/test_utils/test_typing.py:160]: Why is this test being removed?

**Review by guacs — APPROVED** [review]
LGTM.

File: tests/unit/test_utils/test_typing.py (hunk 1 above)
Comment (provinzkraut) [reply tests/unit/test_utils/test_typing.py]: Because I messed something up while rebasing :)

**Pull request merged**